server/src/main/java/org/opensearch/common/geo/GeometryDocValue.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.geo;
+
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.LatLonShape;
+import org.apache.lucene.document.LatLonShapeDocValuesField;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.util.BytesRef;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.index.mapper.GeoShapeIndexer;
+
+import java.util.List;
+
+/**
+ * This class is an OpenSearch Internal representation of lucene {@link LatLonShapeDocValuesField} for GeoShape.
+ *
+ * @opensearch.internal
+ */
+public class GeometryDocValue extends ShapeDocValue {
```

Comment:
```suggestion
public class GeoShapeDocValue extends ShapeDocValue {
```

I think `Geo` for "Geographical" is more accurate and ShapeDocValue can stay. This also keeps the naming consistent w/ the field maper `geo_shape` and `shape` along with the aggregations `geo_`.

Reply:
sure will update this.